### PR TITLE
trigger fsnotify by a meaningful write to the config file

### LINKED ISSUE #3462: File discovery tests often time out
It seems the flaky file discovery test hasn't been fixed. Now the tests often run into the 15s timeout (see travis ci test results, example https://travis-ci.org/prometheus/prometheus/jobs/301314946).

@krasi-georgiev Could you take a look?

### PR DESCRIPTION
**Pull request by krasi-georgiev**
fixes #3462

### CODE REVIEW

File: discovery/file/file_test.go [hunk 1]
```diff
@@ -90,22 +91,21 @@ func testFileSD(t *testing.T, prefix, ext string, expect bool) {
...
 	}
 
-	// File is written with the config so stop draining the discovery channel.
-	// It needs to be before the file closing so that fsnotify triggers a new loop of the discovery service.
+	// Config file is ready so stop draining the discovery channel.
+	// It contains two target groups.
 	close(fileReady)
 	<-drainReady
-	newf.Close()
+	newf.WriteString(" ") // One last meaningless write to trigger fsnotify and a new loop of the discovery service.
```
Comment (Gouthamve) [inline discovery/file/file_test.go]: `newf.Sync()` would do the trick no? And why doesn't `Close()` work here, it also syncs the data to disk, AFAIK.

**Comment by Gouthamve** [comment]
Sorry. Reading it again, the initial writes are drained out and we need to trigger a new FD loop for the changes to be picked up. `Close()` doesn't work for that.

:+1: for the PR.

File: discovery/file/file_test.go (hunk 1 above)
Comment (krasi-georgiev) [reply discovery/file/file_test.go]: I tried `newf.Sync()` and it doesn't trigger a new fsnotify signal

**Comment by krasi-georgiev** [comment]
@grobie PTAL

**Pull request merged**